netket/experimental/driver/tdvp_common.py

```diff
@@ -99,6 +99,7 @@ def __init__(
             self._generator = lambda _: op
         else:
             self._generator = operator
+        self._loss_stats = self.state.expect(self.generator(t0))
```

Comment:
That's not ok. 
This quantity will never be updated later so your callback will just see this value forever.

The generator value is already computed at every (sub)step, for example [here](https://github.com/netket/netket/blob/8bf6fbbdfc6e1e359c613a495a22e6d29f6a4806/netket/experimental/driver/tdvp.py#L137) but it's simply not stored in the driver itself.

You could store during the stage 0 (which corresponds to the current time, which is what you want) by setting it here https://github.com/netket/netket/blob/8bf6fbbdfc6e1e359c613a495a22e6d29f6a4806/netket/experimental/driver/tdvp.py#L151

Reply:
(Please fix also tdvp_schmitt in a similar place)

Reply:
Isn't it the value update just [here](https://github.com/netket/netket/blob/8bf6fbbdfc6e1e359c613a495a22e6d29f6a4806/netket/experimental/driver/tdvp.py#L139C12-L139C23) ? At least, it is the attribute used for the progress bar and the logging. And the value (of <h>) reported in the log changes, so it seems to be updated. 

Or maybe there is something I am missing...

Reply:
Indeed, if we want to avoid supplementary computations, it seems to be the best option